src/read.c

```diff
@@ -789,14 +790,19 @@ static avifResult avifMetaFindOrCreateItem(avifMeta * meta, uint32_t itemID, avi
     assert(itemID != 0);
 
     for (uint32_t i = 0; i < meta->items.count; ++i) {
-        if (meta->items.item[i].id == itemID) {
-            *item = &meta->items.item[i];
+        if (meta->items.item[i]->id == itemID) {
+            *item = meta->items.item[i];
             return AVIF_RESULT_OK;
         }
     }
 
-    *item = (avifDecoderItem *)avifArrayPush(&meta->items);
-    AVIF_CHECKERR(*item != NULL, AVIF_RESULT_OUT_OF_MEMORY);
+    avifDecoderItem ** item_ptr = (avifDecoderItem **)avifArrayPush(&meta->items);
+    AVIF_CHECKERR(item_ptr != NULL, AVIF_RESULT_OUT_OF_MEMORY);
+    *item_ptr = (avifDecoderItem *)avifAlloc(sizeof(avifDecoderItem));
+    AVIF_CHECKERR(*item_ptr != NULL, AVIF_RESULT_OUT_OF_MEMORY);
```

Comment:
I also noticed this issue in my review. So we can't use the `AVIF_CHECKERR` macro here. See the error returns below.

Note: We should not modify `avifMetaDestroy()` to allow null pointers in meta->items unless it's necessary.

Reply:
The reason for my preference is that it will simplify code analysis if a function has no side effects when it fails. Although the Destroy functions typically can clean up all kinds of mess, we should strive to leave the objects in a good, consisent state.